docs/CACHING.md

```diff
@@ -0,0 +1,353 @@
+
+## Metadata Cache
+
+The `metadata-ttl <SECONDS|indefinite|minimal>` flag in `mountOptions` controls the time-to-live (TTL) for cached metadata entries. It can be set to a positive numerical value in seconds, or to one of the pre-configured values of `minimal` (default configuration when not using [Data Cache](#data-cache)) or `indefinite` (metadata entries never expire).
+
+```yaml
+apiVersion: v1
+kind: PersistentVolume
+metadata:
+  name: s3-pv
+spec:
+  mountOptions:
+    - metadata-ttl indefinite # <SECONDS|indefinite|minimal>
+  csi:
+    driver: s3.csi.aws.com
+    # ...
+```
+
+See [Mountpoint's documentation](https://github.com/awslabs/mountpoint-s3/blob/main/doc/CONFIGURATION.md#metadata-cache) for more details about metadata cache.
+
+## Data Cache
+
+Mountpoint supports different types of data caching that you can opt in to accelerate repeated read requests.
+
+### Local Cache
+
+The CSI Driver allows you to configure an [emptyDir](https://kubernetes.io/docs/concepts/storage/volumes/#emptydir) or a [generic ephemeral volume](https://kubernetes.io/docs/concepts/storage/ephemeral-volumes/#generic-ephemeral-volumes) as a local cache.
+The CSI Drivers mounts the provided cache volume to the Mountpoint Pod, and configures Mountpoint to use that volume as local cache.
+
+See [Mountpoint's documentation](https://github.com/awslabs/mountpoint-s3/blob/main/doc/CONFIGURATION.md#local-cache) for more details about local cache.
+
+#### `emptyDir`
+
+You can specify `emptyDir` as cache type in your PV to use an `emptyDir` volume as local cache:
+
+```yaml
+apiVersion: v1
+kind: PersistentVolume
+metadata:
+  name: s3-pv
+spec:
+  # ...
+  csi:
+    driver: s3.csi.aws.com
+    # ...
+    volumeAttributes:
+      bucketName: amzn-s3-demo-bucket
+      cache: emptyDir
+      cacheEmptyDirSizeLimit: 2Gi # optional but highly recommended!
+      cacheEmptyDirMedium: Memory # optional
+```
+
+Both `cacheEmptyDirSizeLimit` and `cacheEmptyDirMedium` are optional, but we highly recommended you to specify a size limit on your cache, it might use all your node's storage otherwise depending on the cluster's configuration. If `cacheEmptyDirMedium` is not specified, the default storage medium will be used.
+
+The `emptyDir` will be unique to the each Mountpoint Pod and won't be shared between other Mountpoint instances.
+
+See [Kubernetes's documentation](https://kubernetes.io/docs/concepts/storage/volumes/#emptydir) for more details about `emptyDir`.
+
+#### `ephemeral`
+
+You can specify `ephemeral` as cache type alongide a StorageClass and storage size in your PV to use an generic ephemeral volume as local cache:
+
+```yaml
+apiVersion: v1
+kind: PersistentVolume
+metadata:
+  name: s3-pv
+spec:
+  # ...
+  csi:
+    driver: s3.csi.aws.com
+    # ...
+    volumeAttributes:
+      bucketName: amzn-s3-demo-bucket
+      cache: ephemeral
+      cacheEphemeralStorageClassName: nvme-ssd # required
```

Comment:
np: `nvme-ssd` -> `gp2` 

At this part of the doc it's not clear where `nvme-ssd` StorageClass is created. Suggesting to use default pre-installed EKS StorageClass `gp2` here or move `nvme-ssd` StorageClass definition before PV definition.